mod_modular_vanilla/hooks/skills/backgrounds/character_background.nut

```diff
@@ -0,0 +1,21 @@
+::ModularVanilla.MH.hook("scripts/skills/character_background", function (q) {
+	// MV: Added
+	// Allows this background to return a different ID during events so that new backgrounds
+	// added by mods work seamlessly with events that check for particular background IDs in their conditions
+	q.MV_getIDForEvent <- function( _eventID )
```

Comment:
Maybe better make this table not a func. A func will encourage catch all approach, i.e. I have a mod adding sailor background but no events for it. So I decide sailor is almost fisherman and write:

```squirrel
function MV_getIDForEvent(_eventID) {
    return "background.fisherman"
}
```

Which means now no mod can create an event for sailor but not fisherman. If it will be table then one will need to write:

```squirrel
MV_IDForEvent = {
    "event.hunt_food": "background.fisherman"
    "event.pirates": "background.fisherman"
    ...
}
```

It has it's downside too though, in if some mod will add an event for fisherman it won't be applied in this case, which might be desired.

Another

Reply:
With the function approach one can define a default ID for a catch-all approach which may be desired. But at the same time it has the capacity to exclude from certain events e.g. your proposed case
```squirrel
function MV_getIDForEvent(_eventID) {
    return "background.fisherman"
}
```
If someone wants to add an event for sailor but not fisherman, he can hook sailor:
```
q.MV_getIDForEvent = @(__original) function( _eventID )
{
   switch (_eventID)
   {
      case "myNewEventForSailorOnly":
      case "someOtherSailorOnlyEvent":
            return this.m.ID;
   }

  return __original();
}
```
Will that not allow covering all possible intended cases?